From cde03b075031efaa6e3f6d69356f077daa6da0dd Mon Sep 17 00:00:00 2001
From: Matthias Koefferlein <matthias@koefferlein.de>
Date: Sun, 17 Dec 2017 15:30:37 +0100
Subject: [PATCH] Replace WindowOkButtonHint, WindowCancelButtonHint

This commit removes symbols which are not available on MacOS
by a code-generation solution.

These symbols are only required for WindowsCE so their
unavailability on other systems is not a big loss.

(cherry picked from commit 28ff493de71cbafa38e214b75dd6874b0226813f)
---
 scripts/mkqtdecl4/mkqtdecl.conf | 2 ++
 scripts/mkqtdecl5/mkqtdecl.conf | 2 ++
 src/gsiqt4/gsiDeclQt_3.cc       | 4 +---
 src/gsiqt5/gsiDeclQt_4.cc       | 4 +---
 4 files changed, 6 insertions(+), 6 deletions(-)

diff --git a/scripts/mkqtdecl4/mkqtdecl.conf b/scripts/mkqtdecl4/mkqtdecl.conf
index 9e87acc..edf48ae 100644
--- a/scripts/mkqtdecl4/mkqtdecl.conf
+++ b/scripts/mkqtdecl4/mkqtdecl.conf
@@ -10,6 +10,8 @@
 # drops enum members
 drop_enum_const "Qt", /WindowType::WindowSoftkeysVisibleHint/
 drop_enum_const "Qt", /WindowType::WindowSoftkeysRespondHint/
+drop_enum_const "Qt", /WindowType::WindowOkButtonHint/ # only available on CE
+drop_enum_const "Qt", /WindowType::WindowCancelButtonHint/ # only available on CE
 
 # --------------------------------------------------------------
 # QtCore
diff --git a/scripts/mkqtdecl5/mkqtdecl.conf b/scripts/mkqtdecl5/mkqtdecl.conf
index 311f08a..d2fd2e5 100644
--- a/scripts/mkqtdecl5/mkqtdecl.conf
+++ b/scripts/mkqtdecl5/mkqtdecl.conf
@@ -10,6 +10,8 @@
 # drops enum members
 drop_enum_const "Qt", /WindowType::WindowSoftkeysVisibleHint/
 drop_enum_const "Qt", /WindowType::WindowSoftkeysRespondHint/
+drop_enum_const "Qt", /WindowType::WindowOkButtonHint/ # only available on CE
+drop_enum_const "Qt", /WindowType::WindowCancelButtonHint/ # only available on CE
 
 # --------------------------------------------------------------
 # QtCore
diff --git a/src/gsiqt4/gsiDeclQt_3.cc b/src/gsiqt4/gsiDeclQt_3.cc
index 0055145..4f74bb9 100644
--- a/src/gsiqt4/gsiDeclQt_3.cc
+++ b/src/gsiqt4/gsiDeclQt_3.cc
@@ -453,9 +453,7 @@ static gsi::Enum<Qt::WindowType> decl_Qt_WindowType_Enum ("Qt_WindowType",
     gsi::enum_const ("WindowStaysOnBottomHint", Qt::WindowStaysOnBottomHint, "@brief Enum constant Qt::WindowStaysOnBottomHint") +
     gsi::enum_const ("WindowCloseButtonHint", Qt::WindowCloseButtonHint, "@brief Enum constant Qt::WindowCloseButtonHint") +
     gsi::enum_const ("MacWindowToolBarButtonHint", Qt::MacWindowToolBarButtonHint, "@brief Enum constant Qt::MacWindowToolBarButtonHint") +
-    gsi::enum_const ("BypassGraphicsProxyWidget", Qt::BypassGraphicsProxyWidget, "@brief Enum constant Qt::BypassGraphicsProxyWidget") +
-    gsi::enum_const ("WindowOkButtonHint", Qt::WindowOkButtonHint, "@brief Enum constant Qt::WindowOkButtonHint") +
-    gsi::enum_const ("WindowCancelButtonHint", Qt::WindowCancelButtonHint, "@brief Enum constant Qt::WindowCancelButtonHint"),
+    gsi::enum_const ("BypassGraphicsProxyWidget", Qt::BypassGraphicsProxyWidget, "@brief Enum constant Qt::BypassGraphicsProxyWidget"),
   "@qt\n@brief This class represents the Qt::WindowType enum");
 
 static gsi::QFlagsClass<Qt::WindowType > decl_Qt_WindowType_Enums ("Qt_QFlags_WindowType",
diff --git a/src/gsiqt5/gsiDeclQt_4.cc b/src/gsiqt5/gsiDeclQt_4.cc
index b768649..6c7fb4b 100644
--- a/src/gsiqt5/gsiDeclQt_4.cc
+++ b/src/gsiqt5/gsiDeclQt_4.cc
@@ -153,9 +153,7 @@ static gsi::Enum<Qt::WindowType> decl_Qt_WindowType_Enum ("Qt_WindowType",
     gsi::enum_const ("MacWindowToolBarButtonHint", Qt::MacWindowToolBarButtonHint, "@brief Enum constant Qt::MacWindowToolBarButtonHint") +
     gsi::enum_const ("BypassGraphicsProxyWidget", Qt::BypassGraphicsProxyWidget, "@brief Enum constant Qt::BypassGraphicsProxyWidget") +
     gsi::enum_const ("NoDropShadowWindowHint", Qt::NoDropShadowWindowHint, "@brief Enum constant Qt::NoDropShadowWindowHint") +
-    gsi::enum_const ("WindowFullscreenButtonHint", Qt::WindowFullscreenButtonHint, "@brief Enum constant Qt::WindowFullscreenButtonHint") +
-    gsi::enum_const ("WindowOkButtonHint", Qt::WindowOkButtonHint, "@brief Enum constant Qt::WindowOkButtonHint") +
-    gsi::enum_const ("WindowCancelButtonHint", Qt::WindowCancelButtonHint, "@brief Enum constant Qt::WindowCancelButtonHint"),
+    gsi::enum_const ("WindowFullscreenButtonHint", Qt::WindowFullscreenButtonHint, "@brief Enum constant Qt::WindowFullscreenButtonHint"),
   "@qt\n@brief This class represents the Qt::WindowType enum");
 
 static gsi::QFlagsClass<Qt::WindowType > decl_Qt_WindowType_Enums ("Qt_QFlags_WindowType",
-- 
2.15.1

